File: Min Cost to Connect All Points - Leetcode 1584/Min Cost to Connect All Points - Leetcode 1584.java
import java.util.*;

public class Solution {
    public int minCostConnectPoints(int[][] points) {
        int n = points.length;
        int totalCost = 0;
        boolean[] visited = new boolean[n];
        PriorityQueue<int[]> minHeap = new PriorityQueue<>(Comparator.comparingInt(a -> a[0]));
        minHeap.offer(new int[]{0, 0}); // cost, pointIndex
        
        while (!minHeap.isEmpty()) {
            int[] edge = minHeap.poll();
            int cost = edge[0];
            int point = edge[1];
            
            if (visited[point]) continue;
            
            visited[point] = true;
            totalCost += cost;
            
            for (int i = 0; i < n; i++) {
                if (!visited[i]) {
                    int newCost = Math.abs(points[point][0] - points[i][0]) + Math.abs(points[point][1] - points[i][1]);
                    minHeap.offer(new int[]{newCost, i});
                }
            }
        }
        
        return totalCost;
    }
}

File: Min Cost to Connect All Points - Leetcode 1584/Min Cost to Connect All Points - Leetcode 1584.cpp
#include <vector>
#include <queue>
#include <cmath>
using namespace std;

class Solution {
public:
    int minCostConnectPoints(vector<vector<int>>& points) {
        int n = points.size();
        int totalCost = 0;
        vector<bool> visited(n, false);
        priority_queue<vector<int>, vector<vector<int>>, greater<vector<int>>> minHeap;
        minHeap.push({0, 0}); // cost, pointIndex
        
        while (!minHeap.empty()) {
            auto edge = minHeap.top();
            minHeap.pop();
            int cost = edge[0];
            int point = edge[1];
            
            if (visited[point]) continue;
            
            visited[point] = true;
            totalCost += cost;
            
            for (int i = 0; i < n; i++) {
                if (!visited[i]) {
                    int newCost = abs(points[point][0] - points[i][0]) + abs(points[point][1] - points[i][1]);
                    minHeap.push({newCost, i});
                }
            }
        }
        
        return totalCost;
    }
};
